src/api/providers/kilocode-ollama.ts

```diff
@@ -140,9 +141,14 @@ export class KilocodeOllamaHandler extends BaseProvider {
 	private ensureClient(): Ollama {
 		if (!this.client) {
 			try {
+				const headers = this.options.ollamaApiKey
+					? { Authorization: this.options.ollamaApiKey } //Yes, this is weird, its not a Bearer token
+					: undefined
+
 				this.client = new Ollama({
 					host: this.options.ollamaBaseUrl || "http://localhost:11434",
-					fetch: fetchWithTimeout(OLLAMA_TIMEOUT_MS),
+					fetch: fetchWithTimeout(OLLAMA_TIMEOUT_MS, headers),
+					headers: headers,
```

Comment:
Why are the headers set twice?

Reply:
Recent versions of the ollama client started allowing for headers, but they don't pass into an overridden fetch function.